src/packet.rs

```diff
@@ -431,7 +431,96 @@ packets! {
+                        let len = self.chunk_data.len() as i32;
+                        try!(<Var<i32> as Protocol>::proto_encode(&len, dst));
+                        for chunk in self.chunk_data.chunks.iter() {
+                            for x in chunk.blocks.iter() {
+                                try!(<u16 as Protocol>::proto_encode(x, dst));
+                            }
+                        }
+                        for chunk in self.chunk_data.chunks.iter() {
+                            try!(dst.write_all(&chunk.block_light));
+                        }
+                        for chunk in self.chunk_data.chunks.iter() {
+                            match chunk.sky_light {
+                                Some(xs) => try!(dst.write_all(&xs)),
+                                None => {}
+                            }
+                        }
+                        match self.chunk_data.biomes {
+                            Some(xs) => try!(dst.write_all(&xs)),
+                            None => {}
+                        }
+                        Ok(())
+                    }
+                    #[allow(unused_variables)]
+                    fn decode(mut src: &mut Read, len: usize) -> io::Result<ChunkData> {
+                        use std::num::Int;
+                        use types::Chunk;
+
+                        let x = try!(<i32 as Protocol>::proto_decode(src));
+                        let z = try!(<i32 as Protocol>::proto_decode(src));
+                        let continuous = try!(<bool as Protocol>::proto_decode(src));
+                        let mask = try!(<u16 as Protocol>::proto_decode(src));
+                        let len = try!(<Var<i32> as Protocol>::proto_decode(src));
+
+                        let num_chunks = mask.count_ones();
+                        let mut chunks = Vec::new();
+                        for _ in 0..num_chunks {
+                            chunks.push(Chunk::empty());
+                        }
+                        let mut column = ChunkColumn{
+                            chunks: chunks,
+                            biomes: None
+                        };
+                        for chunk in column.chunks.iter_mut() {
+                            for x in chunk.blocks.iter_mut() {
+                                *x = try!(<u16 as Protocol>::proto_decode(src));
+                            }
+                        }
+                        for chunk in column.chunks.iter_mut() {
+                            // We use this instead of read_exact because it's an array, Vec is useless here.
+                            for x in chunk.block_light.iter_mut() {
+                                *x = try!(<u8 as Protocol>::proto_decode(src));
+                            }
+                        }
+                        // FIXME(toqueteos): Tricky one, only sent if player is in Overworld.
```

Comment:
We might have to save the raw chunk column data, and give it a decode method that takes the dimension.